api/envoy/extensions/filters/http/grpc_stats/v3/config.proto

```diff
@@ -60,6 +61,14 @@ message FilterConfig {
   // <envoy_v3_api_field_extensions.filters.http.grpc_stats.v3.FilterConfig.individual_method_stats_allowlist>` the same way
   // request_message_count and response_message_count works.
   bool enable_upstream_stats = 4;
+
+  // If true, the filter will replace dots in the grpc_service_name with underscores before emitting
+  // the metrics. Only works when :ref:`stats_for_all_methods
+  // <envoy_v3_api_field_extensions.filters.http.grpc_stats.v3.FilterConfig.stats_for_all_methods>`
+  // is set to true. It could cause metrics to be merged if the edited service name conflicts with
+  // an existing service. For example there are both service "foo.bar" & "foo_bar" running.
+  // By default this is set as false.
+  bool replace_dots_in_grpc_service_name = 5;
```

Comment:
What's the use case of this, i.e. why dots are problematic here for your stats?

Reply:
The gRPC stats regex Envoy is using now assuming no dots in the gRPC service name. https://github.com/envoyproxy/envoy/issues/16384 Sorry I forgot to add the issue link in the PR description. Just added that.

Reply:
Added.